### PR TITLE
Implement code formatter

### LINKED ISSUE #6: Implement Linter
Hi @cloudhead,

Wanted to know if you thought about implementing some kind of linter like ESLint to enforce code standards and style?
If you agree I could propose some standard rules that apply to the current code and enforce it for future contributions.

Cheers

### PR DESCRIPTION
**Pull request by sebastinez**
This PR closes #6 

I used `prettier` and `prettier-plugin-svelte` to implement a code formatter.
The configuration is pretty simple and originally from `radicle-upstream` with some minor changes for this repo.

To be able to use GitHub CI I created a very bare GitHub Action, and the core is in the shell script `scripts/format`

Let us discuss here which format options are the correct ones for this repo
https://prettier.io/docs/en/options.html

### CODE REVIEW

**Comment by cloudhead** [comment]
I've gone through the formatted code, and I think it does a good job on the `.ts` files, but not the HTML in the .svelte files, as it wraps or unwraps lines seemingly randomly, hurting readability.

So as a first step I'd propose these settings:

```diff
diff --git a/prettier.config.js b/prettier.config.js
index e08b9ff..bee6d04 100644
--- a/prettier.config.js
+++ b/prettier.config.js
@@ -2,10 +2,10 @@ module.exports = {
   "arrowParens": "avoid",
   "tabWidth": 2,
   "semi": true,
-  "singleQuote": true,
+  "singleQuote": false,
   "quoteProps": "preserve",
-  "trailingComma": "es5",
+  "trailingComma": "all",
```

and a `.prettierignore` with

```
*.html
*.svelte
*.css
```

**Comment by sebastinez** [comment]
@cloudhead force pushed the latest changes like we talked about

File: scripts/check-format [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+# Checks local /node_modules,
+# if not found will download that package and execute it.
+# Does not modify files, informs bad code format
+npx eslint .
```
Comment (cloudhead) [inline scripts/check-format:6]: Until we've linted the codebase, we should keep this commented out.

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (cloudhead) [inline .eslintignore:1]: Missing a new-line here and in the other file too

File: scripts/check-format (hunk 1 above)
Comment (sebastinez) [reply scripts/check-format:6]: Eventually we could keep it like this and disable the CI in the .github/workflows folder, so it doesn't fail on every push

Comment (cloudhead) [reply scripts/check-format:6]: Yeah that's fine too.

Comment (sebastinez) [reply scripts/check-format:6]: [We can disable the workflow from the GitHub UI until everything is linted.](https://docs.github.com/es/actions/managing-workflow-runs/disabling-and-enabling-a-workflow#disabling-and-enabling-workflows-with-the-github-ui)

What do you think?

**Comment by cloudhead** [comment]
Hmm weird, build is failing for platform reasons :thinking:

**Comment by sebastinez** [comment]
> Hmm weird, build is failing for platform reasons thinking

It seems like an issue to how the fsevents package got compiled, it wants macOS darwin instead of linux, 
```sh
5:10:17 PM: npm ERR! notsup Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
```
Eventually I could remove the package-lock.json file I commited, and see if this helps

**Comment by sebastinez** [comment]
I think to finish this PR you should commit a new package.lock.json based on the new packages used.
Since I can't create one on macOS.

**Comment by cloudhead** [comment]
> @cloudhead I think to finish this PR you should commit a new package.lock.json based on the new packages used.
> Since I can't create one on macOS.

You mean on `master`?

**Comment by sebastinez** [comment]
Yes this should do the trick. Or I could remove the packages from packages.json and you could install them

**Comment by cloudhead** [comment]
Ok done, I think. Or does this require a completely fresh lock file? Trying to avoid updating all dependencies though.

**Comment by sebastinez** [comment]
I'll rebase this branch without updating the lock file but including the new dependencies, and then you could do a `npm install` to update the lock file

**Comment by cloudhead** [comment]
> I'll rebase this branch without updating the lock file but including the new dependencies, and then you could do a `npm install` to update the lock file

This should be done already - I added the dependencies to master.

**Comment by sebastinez** [comment]
Okay great! Then I'll remove both package.json and lock file from this PR

**Review by cloudhead — APPROVED** [review]

**Pull request merged**